include/envoy/network/connection.h

```diff
@@ -153,15 +164,20 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   virtual void write(Buffer::Instance& data) PURE;
 
   /**
-   * Set a soft limit on the size of the read buffer prior to flushing to further stages in the
+   * Set a soft limit on the size of buffers for the connection.
+   * For the read buffer, this limits the bytes read prior to flushing to further stages in the
    * processing pipeline.
+   * For the write buffer, it sets watermarks.  When enough data is buffered it triggers a call to
+   * onAboveWriteBufferHighWatermark, which disables reads on the socket which is funneling data to
```

Comment:
doc nit: onAboveWriteBufferHighWatermark() doesn't necessarily disable reads. It's obv up to the the code that receives the callback to do what they want (potentially just stats, etc.). I might clarify this slightly since it seems like this code will somehow disable reads.